src/hotspot/share/opto/superword.cpp

```diff
@@ -1511,130 +1499,110 @@ SuperWord::SplitStatus SuperWord::split_pack(const char* split_name,
 }
 
 template <typename SplitStrategy>
-void SuperWord::split_packs(const char* split_name,
-                            SplitStrategy strategy) {
+void PackSet::split_packs(const char* split_name,
+                          SplitStrategy strategy) {
   bool changed;
   do {
     changed = false;
     int new_packset_length = 0;
-    for (int i = 0; i < _packset.length(); i++) {
-      Node_List* pack = _packset.at(i);
+    for (int i = 0; i < _packs.length(); i++) {
+      Node_List* pack = _packs.at(i);
       assert(pack != nullptr && pack->size() >= 2, "no nullptr, at least size 2");
       SplitTask task = strategy(pack);
       SplitStatus status = split_pack(split_name, pack, task);
       changed |= !status.is_unchanged();
       Node_List* first_pack = status.first_pack();
       Node_List* second_pack = status.second_pack();
-      _packset.at_put(i, nullptr); // take out pack
+      _packs.at_put(i, nullptr); // take out pack
       if (first_pack != nullptr) {
         // The first pack can be put at the current position
         assert(i >= new_packset_length, "only move packs down");
-        _packset.at_put(new_packset_length++, first_pack);
+        _packs.at_put(new_packset_length++, first_pack);
       }
       if (second_pack != nullptr) {
         // The second node has to be appended at the end
-        _packset.append(second_pack);
+        _packs.append(second_pack);
       }
     }
-    _packset.trunc_to(new_packset_length);
+    _packs.trunc_to(new_packset_length);
   } while (changed);
 
 #ifndef PRODUCT
   if (is_trace_superword_packset()) {
     tty->print_cr("\nAfter %s", split_name);
-    print_packset();
+    print();
   }
 #endif
 }
 
 // Split packs at boundaries where left and right have different use or def packs.
 void SuperWord::split_packs_at_use_def_boundaries() {
```

Comment:
Yes, let's consider that for a future RFE.

But for now I can say this:
We moved the implementation of `split / filter` to the packset, and that makes sense: we can hide the packset internals, and don't have them spill out to the SuperWord code. We can just call `_packset.split_packs` with a `split_strategy`. But defining the split strategy itself depends on other SuperWord components, and so I think conceptually they belong with SuperWord. They query `reductions`, `dependency graph`, `implemented`, `AlignmentSolution`, `profitable`, etc.
Sure, we can just pass a SuperWord reference, but that does not really seem right to me either. For me, the SuperWord class is there to manage the interface between all the components, and try to avoid passing components to other components, wherever possible. So I would rather have a list of methods in SuperWord, and each such method defines how the components interact (e.g. packset and alignment, packset and AlignmentSolution, pairset and packset, ...).
But we can discuss this further, and maybe come up with an even better solution. My hope is just that we separate the components as much as possible, so that we know that only a handful of them interact at a given point. That makes the whole beast more managable.

Reply:
That's a valid point. Let's have another look at `SuperWord`, `PairSet` and `PackSet` once the code stabilizes to check whether it needs improvements or not by moving some code to the `*Set` classes.